site/student-lectures/cultural-algorithms/cultural-algorithms.rst

```diff
@@ -0,0 +1,255 @@
+*******************
+Cultural Algorithms
+*******************
+
+Introduction
+============
+* It is a branch of evolutionary computation
+* Unlike genetic algorithms, cultural algorithms have not only **population space** but also **belief space**
+* Here shows the components of cultural algorithms and its structure [#]_:
+
+.. figure:: ca_components.png
+    :align: center
+
+
+* Object() is the fitness function
+* The most important part of the cultural algorithm is:
+
+    * Accept() picks top n individuals from population
+    * Update() updates belief space based on the individuals pick from Accept()
+    * Influence() is to use the knowledge in belief space to update population
+
+
+
+Example Problem -- Minimization
+===============================
+* Find a vector :math:`x` to minimize :math:`f(x)`
+* A simple example is :math:`f(x) = x_1^2 + x_2^2`
+* So the problem is to find a vector :math:`(x_1, x_2)` to minimise :math:`f(x)`
+
+* Fitness function:
+
+.. literalinclude:: ca.py
+    :language: python
+    :lineno-match:
+    :pyobject: fitness_function
```

Comment:
If you change the code to NOT be a function like mentioned above. then you can add tags to your code. Have a look at some of my rst files with code to have a look at how I do it.